cloud/amazon/public/resources/launchconfiguration.go

```diff
@@ -182,9 +181,12 @@ func (r *Lc) Apply(actual, expected cloud.Resource, immutable *cluster.Cluster)
 					if instance.PublicIpAddress != nil {
 						privip = *instance.PrivateIpAddress
 						pubip = *instance.PublicIpAddress
-						immutable.Values.ItemMap["INJECTEDMASTER"] = fmt.Sprintf("%s:%s", privip, immutable.KubernetesAPI.Port)
-						immutable.KubernetesAPI.Endpoint = pubip
+
+						providerConfig := immutable.ProviderConfig()
+						providerConfig.Values.ItemMap["INJECTEDMASTER"] = fmt.Sprintf("%s:%s", privip, immutable.ProviderConfig().KubernetesAPI.Port)
+						providerConfig.KubernetesAPI.Endpoint = pubip
```

Comment:
Just wondering, if we're injecting Private IP here, why is this using public IP? So we can access it outside the cluster?

Reply:
This is unchanged from before, just using provider config. I think this is how the nodes talk to the master so it doesnt need to be (and probably shouldnt be) private